core/capabilities/registry.go

```diff
@@ -20,6 +22,34 @@ type Registry struct {
 	lggr logger.Logger
 	m    map[string]capabilities.BaseCapability
 	mu   sync.RWMutex
+
+	// Mutex for getLocalNode, we don't want to lock the entire registry
+	glnMu        sync.RWMutex
+	getLocalNode func(ctx context.Context) (capabilities.Node, error)
+}
+
+var _ core.CapabilitiesRegistry = (*Registry)(nil)
+
+func (r *Registry) SetGetLocalNodeFunc(f func(ctx context.Context) (capabilities.Node, error)) {
+	r.glnMu.Lock()
+	defer r.glnMu.Unlock()
+	r.getLocalNode = f
+}
+
+func (r *Registry) GetLocalNode(ctx context.Context) (capabilities.Node, error) {
```

Comment:
Arguably you should do all this in the launcher: that'll allow you to make use of the fact that these are both in the same package (i.e. you can use private setters), plus it should mean not having to resolve the value in a background routine as well; you can just set it once we have the right value.

Reply:
I'm in the middle of this PR: https://github.com/smartcontractkit/chainlink/pull/13750 which moves LocalNode + ConfigForCapability to the registrysyncer package, so it might be worth discussing how this all matches up correctly.

One option is to pass the configProvider to the launcher when syncing (probably by attaching it to the State object?) and then setting that on the capabilities registry.

Reply:
I dont think i fully understand what changes i need to make, but I didnt like the fact that I had to poll to resolve the local node value here. If we can avoid that by moving this logic into the launcher i'd much prefer that